app/controllers/tasks_controller.rb

```diff
@@ -26,6 +26,31 @@ def index
     @tasks = Task.all
   end
 
+  def edit
+    @task = Task.find_by(requester_id: current_user.id)
+    @task = Task.find(params[:id])
+    @task_categories = TaskCategory.all
+  end
```

Comment:
Because we are dealing with the general list of task from the Task controller, had to put that line to narrow down to the current user to ensure the task being called belongs to the user.

Several users could have the same task ID, line 30 is to first call the current signed in requester then call the task associated with that requester.